src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -127,23 +153,23 @@ internal static bool IsEnabled()
                 if (activity != null)
                 {
                     s_diagnosticListener.StopActivity(activity, new ActivityStopData(
-                        responseTask?.Status == TaskStatus.RanToCompletion ? responseTask.Result : null,
+                        response,
                         // If request is failed or cancelled, there is no response, therefore no information about request;
                         // pass the request in the payload, so consumers can have it in Stop for failed/canceled requests
                         // and not retain all requests in Start
                         request,
-                        responseTask?.Status ?? TaskStatus.Faulted));
+                        responseTask?.Status ?? taskStatus));
```

Comment:
If the sync API throws due to cancellation, isn't this going to report Faulted rather than Canceled?  Is that desired?

Reply:
(It seems like you should change the `TaskStatus taskStatus = TaskStatus.Faulted;` to instead be `TaskStatus taskStatus = TaskStatus.RanToCompletion;`.  Then in the `catch (OperationCanceledException)` block, you'd set it to Canceled, and in the `catch (Exception)` block, you'd set it to Faulted.  Then in the finally block where you need the status, you just always use taskStatus.  And then you also can get rid of responseTask and just await the SendAsync directly.)